libs/@guardian/source-react-components-development-kitchen/src/file-input/stringifyFileBase64.ts

```diff
@@ -0,0 +1,30 @@
+export const stringifyFileBase64 = (file: File): Promise<string> =>
+	new Promise((resolve, reject) => {
+		const reader = new FileReader();
+		reader.addEventListener(
+			'load',
+			() => {
+				// remove data:*/*;base64, from the start of the base64 string
+				const fileAsBase64 = reader.result?.toString().split(';base64,')[1];
+
+				if (fileAsBase64) {
+					resolve(fileAsBase64);
+				} else {
+					reject(
+						new Error(
+							'Sorry there was a problem with the file you uploaded above. Check the size and type. We only accept images, pdfs and .doc or .docx files',
```

Comment:
Does the bit about .doc and .docx need to be removed from lines 15 and 25, or added to the list of accepted files?

Reply:
I've updated the way this works. The component now takes a `validFileType` which defaults to `['image/png', 'image/jpeg', 'image/jpg']`, and an optional `maxFileSize`, which are both used for additional information, and to construct the appropriate error message when an incorrect file is uploaded.